bigquery_etl/cli/stage.py

```diff
@@ -203,15 +203,12 @@ def _udf_dependencies(artifact_files):
     for udf_file in udf_files:
         # all referenced UDFs need to be deployed in the same stage project due to access restrictions
         raw_routine = RawRoutine.from_file(udf_file)
-        udfs_to_publish = accumulate_dependencies([], raw_routines, raw_routine.name)
-
-        for dependency in udfs_to_publish:
-            dataset, name = dependency.split(".")
-            file_path = (
-                raw_routine.filepath.parent.parent.parent / dataset / name / UDF_FILE
-            )
-            if file_path not in artifact_files:
-                artifact_files.add(file_path)
```

Comment:
This seems like it was a bug, modifying the passed in `artifact_files` set rather than adding to the `udf_dependencies` set that gets returned.